var/spack/repos/builtin/packages/amrvis/package.py

```diff
@@ -0,0 +1,239 @@
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.  #
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Amrvis(MakefilePackage):
+    """Amrvis is a visualization package specifically designed to
+       read and display output and profiling data from codes built
+       on the AMReX framework.
+    """
+
+    homepage = "https://github.com/AMReX-Codes/Amrvis"
+    url      = "https://github.com/AMReX-Codes/Amrvis.git"
+
+    version('master',
+            git='https://github.com/AMReX-Codes/Amrvis.git', tag='master')
+
+    variant(
+        'dims',
+        default='3',
+        values=('1', '2', '3'),
+        multi=False,
+        description='Number of spatial dimensions'
+    )
+    variant(
+        'prec',
+        default='DOUBLE',
+        values=('FLOAT', 'DOUBLE'),
+        multi=False,
+        description='Floating point precision'
+    )
+    variant('mpi', default=False, description='Enable MPI parallel support')
+    variant('debug', default=False, description='Enable debugging features')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('libsm')
+    depends_on('libice')
+    depends_on('libxpm')
+    depends_on('libx11')
+    depends_on('libxt')
+    depends_on('libxext')
+    depends_on('motif')
+
+    # Only doing gcc and clang at the moment
+    # Intel currently fails searching for mpiicc, mpiicpc, etc
+    conflicts('%intel',
+              msg='Amrvis currently only builds with gcc and clang')
+    conflicts('%cce',
+              msg='Amrvis currently only builds with gcc and clang')
+    conflicts('%nag',
+              msg='Amrvis currently only builds with gcc and clang')
+    conflicts('%pgi',
+              msg='Amrvis currently only builds with gcc and clang')
+    conflicts('%xl',
+              msg='Amrvis currently only builds with gcc and clang')
+    conflicts('%xl_r',
+              msg='Amrvis currently only builds with gcc and clang')
```

Comment:
You could store this `msg` in a variable so you don't have to keep repeating yourself. Even better, you could add these `conflicts` in a for-loop for each compiler.

Reply:
Will do.